### ISSUE TITLE
The data could not be converted to UTF-8

### ISSUE THREAD

**Issue by miladghiravani** [simplepie/simplepie#306]
Hi!
I use simplepie for get rss. I create a file (test1.php) in simplepie script in the path (simplepie/demo) with this code:

``` php
header('Content-type:text/html; charset=utf-8');

include_once('../autoloader.php');
include_once('../idn/idna_convert.class.php');

$feed = new SimplePie();
$feed->set_feed_url('http://pop-music.ir/feed');
$feed->force_feed(true);
$feed->set_input_encoding('UTF-8');
$feed->set_output_encoding('UTF-8');
$feed->init();
$feed->handle_content_type();
echo $feed->get_title();

if ($feed->error())
{
    echo '' . "\r\n";

        // ... and display it.
        echo '<p>' . htmlspecialchars($feed->error()) . "</p>\r\n";
    echo ' . "\r\n";
}
?>
```

But the script gives the following error:

<b>The data could not be converted to UTF-8. You MUST have either the iconv or mbstring extension installed. Upgrading to PHP 5.x (which includes iconv) is highly recommended.</b>

My host is upgraded for PHP 5.2.17. 
![untitled-4](https://f.cloud.github.com/assets/5067331/837108/0e70ef64-f300-11e2-9acf-7df5b5e7f43c.jpg)

When I check url rss like this <b>http://pop-music.ir/feed</b> on my script, I get error but when check this url in demo simplepie <a href="http://www.simplepie.org/demo/?feed=http://pop-music.ir/feed">view demo this url</a> I can view rss!!!!

Please Help me:(

**Comment by EmBitz** [comment]
@webhostingtop 

Would be nice if you could share how to fix this parser problem. I'm facing the same issues.
Thanks.

**Comment by webhostingtop** [comment]
@EmBitz 
Have you tried latest Version 1.4.2 ? Check http://simplepie.org/downloads/ and let me know if you still have the same issue.

**Comment by EmBitz** [comment]
Thanks for your reply.

Version 1.4.2 same problem:
 
PHP Notice: The data could not be converted to UTF-8. You MUST have either the iconv or mbstring extension installed. Upgrading to PHP 5.x (which includes iconv) is highly recommended. in wp-includes/class-simplepie.php on line 1421

**Comment by skyzyx** [comment]
Well, the answer is in the message itself. You MUST have either the **iconv** or **mbstring** extension installed. Do you have those?

Have you run the compatibility test?

**Comment by JanPetterMG** [comment]
Please update to 1.4.3, as *a lot* has changed since 1.4.2.
Now UConverter is supported, and the PHP notice you're referring to is replaced by an dynamically built notice based on what you're missing.
There was an "empty feed" issue, resulting in an incorrect "could not be converted to utf-8" notice, that also got fixed in 1.4.3.
If the problem still occurs, it would be a lot easier for us to debug, if you could post the feed URL.

**Comment by szabacsik** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  andras.szabacsik@gmail.com
    Domain szabacsik.hu has exceeded the max emails per hour (52/50 (104%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o7.sgmail.github.com ([167.89.101.198]:19463)
	by hardcube.23net.hu with esmtps (TLSv1:DHE-RSA-AES128-SHA:128)
	(Exim 4.87)
	(envelope-from <bounces+848413-d5b6-andras=szabacsik.com@sgmail.github.com>)
	id 1dPM6j-003dcX-Jw
	for andras@szabacsik.com; Mon, 26 Jun 2017 06:58:42 +0200
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=7hnHdyVb7lqyzQZFCBmtJZ4j6A8=; b=Q+vzTcZMydQGt2M4
	alPifE0ljm1T7ASZslYTFlgM2+Jxhnwo5JOX791C29gCJSHaec8vHLn7vyzpp0DM
	RizBodTmi/flu3Wg0AazZTdyA4JiK3Vy6satEvfAuFnJeRDxzZCl+a6K+H9EYesC
	KjXuoySU1ffYhEpRlnJozeldxr0=
Received: by filter0438p1mdw1.sendgrid.net with SMTP id filter0438p1mdw1-25417-59509456-2F
        2017-06-26 04:57:58.724846317 +0000 UTC
Received: from github-smtp2b-ext-cp1-prd.iad.github.net (github-smtp2b-ext-cp1-prd.iad.github.net [192.30.253.17])
	by ismtpd0002p1iad1.sendgrid.net (SG) with ESMTP id yTtKGg4ZT7aVE4zMtglG9Q
	for <andras@szabacsik.com>; Mon, 26 Jun 2017 04:57:58.606 +0000 (UTC)
Date: Sun, 25 Jun 2017 21:57:58 -0700
